crates/subspace-service/src/subspace_dsn.rs

```diff
@@ -0,0 +1,62 @@
+use futures::StreamExt;
+use parity_scale_codec::Encode;
+use sc_consensus_subspace::{ArchivedSegmentNotification, SubspaceLink};
+use sp_core::traits::SpawnEssentialNamed;
+use sp_runtime::traits::Block as BlockT;
+use subspace_networking::{CreationError, PUB_SUB_ARCHIVING_TOPIC};
+use tracing::{error, info, trace};
+
+/// Start an archiver that will listen for archived segments and send it to DSN network using
+/// pub-sub protocol.
+pub async fn start_subspace_dsn_node<Block, Spawner>(
+    subspace_link: &SubspaceLink<Block>,
+    networking_config: subspace_networking::Config,
+    spawner: Spawner,
+) -> Result<(), CreationError>
+where
+    Block: BlockT,
+    Spawner: SpawnEssentialNamed,
+{
+    trace!("Subspace networking starting.");
+
+    let (node, node_runner) = subspace_networking::create(networking_config).await?;
+
+    info!("Subspace networking initialized: Node ID is {}", node.id());
+
+    spawner.spawn_essential(
+        "node-runner",
+        Some("subspace-networking"),
+        Box::pin(node_runner.run()),
+    );
+
+    let mut archived_segment_notification_stream = subspace_link
+        .archived_segment_notification_stream()
+        .subscribe();
+
+    spawner.spawn_essential(
+        "archiver",
+        Some("subspace-networking"),
+        Box::pin(async move {
+            trace!("Subspace DSN archiver started.");
+
+            while let Some(ArchivedSegmentNotification {
+                archived_segment, ..
+            }) = archived_segment_notification_stream.next().await
+            {
+                trace!("ArchivedSegmentNotification received");
```

Comment:
Probably nicer to add a target to the log like `trace!(target: "dsn", "Subspace DSN archiver started.");` to achieve some flexibility?